src/LaTeXStrings.jl

```diff
@@ -29,21 +29,54 @@ struct LaTeXString <: AbstractString
     s::String
 end
 
+# the only point of using LaTeXString to represent equations, since
+# IJulia doesn't support LaTeX output other than equations, so add $'s
+# around the string if they aren't there (ignoring \$ and %$)
+_maybe_wrap_equation(s) = occursin(r"[^\\%]\$|^\$", s) ? s : string('\$', s, '\$')
+
 """
     latexstring(args...)
 
 Similar to `string(args...)`, but generates a `LaTeXString` instead of a `String`.
 """
 latexstring(args...) = latexstring(string(args...))
 function latexstring(s::String)
-    # the only point of using LaTeXString to represent equations, since
-    # IJulia doesn't support LaTeX output other than equations, so add $'s
-    # around the string if they aren't there (ignoring \$)
-    return occursin(r"[^\\]\$|^\$", s) ? LaTeXString(s) :  LaTeXString(string('\$', s, '\$'))
+    return LaTeXString(_maybe_wrap_equation(s))
 end
 latexstring(s::AbstractString) = latexstring(String(s))
 
-macro L_str(s, flags...) latexstring(s) end
+if isdefined(Meta, :parseatom)
+    const parseatom = Meta.parseatom
+else
+    parseatom(s, i; filename=nothing) = Meta.parse(s, i, greedy=false)
+end
+
+macro L_str(s, flags...)
+    s = _maybe_wrap_equation(s)
+    i = firstindex(s)
+    buf = IOBuffer(maxsize=ncodeunits(s))
+    ex = Expr(:string)
+    while i <= ncodeunits(s)
+        c = @inbounds s[i]
+        if c === '%'
+            i = nextind(s, i)
+            i > ncodeunits(s) && break
```

Comment:
Does this mean that if a `%` is the last character of the string then it will be lost?

Reply:
Yes, good catch! Will fix that!